yarl/_path.py

```diff
@@ -39,3 +40,32 @@ def normalize_path(path: str) -> str:
 
     segments = path.split("/")
     return prefix + "/".join(normalize_path_segments(segments))
+
+
+def calculate_relative_path(target: str, base: str) -> str:
+    """Return the relative path between two other paths.
+
+    If the operation is not possible, raise ValueError.
+    """
+
+    target = target or "/"
+    base = base or "/"
+
+    target_path = PurePosixPath(target)
+    base_path = PurePosixPath(base)
+
+    if not base.endswith("/"):
+        base_path = base_path.parent
+
+    for step, path in enumerate([base_path] + list(base_path.parents)):
+        if target_path.is_relative_to(path):
+            break
+        elif path.name == "..":
+            raise ValueError(f"'..' segment in {str(base_path)!r} cannot be walked")
+    else:
+        raise ValueError(
+            f"{str(target_path)!r} and {str(base_path)!r} have different anchors"
+        )
+    offset = len(path.parts)
+    parts = [".."] * step + list(target_path.parts)[offset:]
+    return str(PurePosixPath(*parts))
```

Comment:
```suggestion
    return str(PurePosixPath(*("..",) * step, *target_path.parts[offset:]))
```

I think you can avoid creating all the lists